### PR TITLE
Added "Hue Shift" Effect

### PR DESCRIPTION
**Pull request by SRG27**
imag

### CODE REVIEW

File: ChaosMod/Effects/db/Screen/Shaders/ScreenShaderHueShift.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	float3x3 result = mul(mat, colMat);
+
+    return float4(saturate(result[0][0]), saturate(result[1][1]), saturate(result[2][2]), 1.);
+}
+)SRC";
+
+static void OnStart()
+{
+	Hooks::OverrideShader(EOverrideShaderType::LensDistortion, ms_szShaderSrcPrefix + std::to_string(g_Random.GetRandomFloat(0.f, 360.f)) + ms_szShaderSrcSuffix);
```
Comment (OnlyRealNubs) [inline ChaosMod/Effects/db/Screen/Shaders/ScreenShaderHueShift.cpp:43]: Why not generate the random number in the shader source?

Comment (OnlyRealNubs) [reply ChaosMod/Effects/db/Screen/Shaders/ScreenShaderHueShift.cpp:43]: Pongo knows more about this than I do. So maybe ask him, but if you can't do it. I guess just stick with what you got lol.

**Comment by SRG27** [comment]
Accidentally closed

**Comment by pongo1231** [comment]
Conflicts need to be resolved before this can be merged. Looks fine to me though.

**Comment by SRG27** [comment]
Messed up this branch too much to fix it. Will open another PR for this

**Pull request closed without merging**